src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.Helpers.cs

```diff
@@ -13,35 +13,6 @@ public static partial class JsonSerializer
         // Members accessed by the serializer when serializing.
         private const DynamicallyAccessedMemberTypes MembersAccessedOnWrite = DynamicallyAccessedMemberTypes.PublicProperties | DynamicallyAccessedMemberTypes.PublicFields;
 
-        private static void WriteCore<TValue>(
-            Utf8JsonWriter writer,
-            in TValue value,
-            Type inputType,
-            JsonSerializerOptions options)
-        {
-            Debug.Assert(writer != null);
-
-            //  We treat typeof(object) special and allow polymorphic behavior.
-            if (inputType == JsonTypeInfo.ObjectType && value != null)
-            {
-                inputType = value.GetType();
-            }
-
-            WriteStack state = default;
-            JsonConverter jsonConverter = state.Initialize(inputType, options, supportContinuation: false);
-
-            try
-            {
-                bool success = WriteCore(jsonConverter, writer, value, options, ref state);
-                Debug.Assert(success);
-            }
-            catch
-            {
-                state.DisposePendingDisposablesOnException();
-                throw;
-            }
-        }
-
         private static bool WriteCore<TValue>(
```

Comment:
nit: "WriteCore" and "SerializeUsingMetadata" have different terminology. Suggest making them consistent.